File: 2620. Counter/solution1.js
/**
 * @param {number} n
 * @return {Function} counter
 */
var createCounter = function(n) {
    let n2 = n;
    return function() {
        return n2++;
    };
};

/** 
 * const counter = createCounter(10)
 * counter() // 10
 * counter() // 11
 * counter() // 12
 */

File: Easy/374. Guess Number Higher or Lower/solution1.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        upperBound = n
        lowerBound = 1

        while (guess(n) != 0):
            if (guess(n) == -1):
                upperBound = n
                n = math.trunc((n-lowerBound)/2)
            else:
                lowerBound = n
                n += math.trunc((upperBound-n)/2)
        return n